docs/api/hyperparameters.rst

```diff
@@ -0,0 +1,106 @@
+.. _Hyperparameters:
+
+Hyperparameters
+===============
+ConfigSpace contains
+:func:`~ConfigSpace.api.types.float.Float`,
+:func:`~ConfigSpace.api.types.int.Int`
+and :func:`~ConfigSpace.api.types.categorical.Categorical` hyperparamters, each with their own customizability.
+
+For :func:`~ConfigSpace.api.types.float.Float` and :func:`~ConfigSpace.api.types.int.Int`, you will find their
+interface much the same, being able to take the same :ref:`distributions <Distributions>` while :func:`~ConfigSpace.api.types.categorical.Categorical` can take weights or be ordered.
+
+These are all convenience functions that construct the more complex :ref:`hyperparameter classes <Advanced_Hyperparameters>` which make up the backbone of what's possible.
```

Comment:
The header now drops the mention of ordinal hyperparameters, is that on purpose? Also, maybe it would be good to briefly mention that there are simple and advanced hyperparameter classes, and how they differ.

Reply:
So I guess that is capture in mentioning "...`Categorical` can take weights or be **ordered**" but maybe we should emphasise this more.

Will do on the second point on mentioning how they differ.